production/training/docker/Dockerfile.Redhat

```diff
@@ -0,0 +1,73 @@
+FROM registry.access.redhat.com/rhel7/rhel
+
+WORKDIR /home/downloads
+
+MAINTAINER Almond Admin <root@almond.stanford.edu>
+
+# args can be set using --build-args, e.g.:
+# --build-arg ALMOUND_CLOUD_VERSION=v1.4.0
+ARG ALMOND_CLOUD_VERSION=v1.5.1
+
+# add redhat repos
+RUN yum-config-manager \
+	--enable rhel-server-rhscl-7-rpms  \
+	--enable rhel-7-server-optional-rpms 
+
+# install basic tools
+RUN yum -y install  wget curl vim net-tools
+
+# add epel repo 
+RUN wget http://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+RUN rpm -ivh epel-release-latest-7.noarch.rpm
+
+# install python3
+RUN yum -y install @development \
+	rh-python36 \
+	rh-python36-numpy \
+	rh-python36-scipy \
+	rh-python36-python-tools  \
+	rh-python36-python-six \
+	rh-python36-PyYAML \
+        python36-pip \
+	python36-devel
+
+RUN scl enable rh-python36 bash
+
+# install nodejs 10.x and yarn
+RUN curl -sL https://rpm.nodesource.com/setup_10.x | bash -
+RUN curl -sL https://dl.yarnpkg.com/rpm/yarn.repo | tee /etc/yum.repos.d/yarn.repo
+RUN yum -y install nodejs yarn
+
+# install cvc4
+RUN wget https://almond-static.stanford.edu/test-data/cvc4-1.6-x86_64-linux-opt -O /usr/bin/cvc4
+RUN chmod 755 /usr/bin/cvc4
+
+# install other dependencies
+RUN yum -y install \
+	GraphicsMagick \
+	java-1.8.0-openjdk-devel ant \
+	hunspell-devel
+
+# install decanlp
+RUN pip3 install --user 'git+https://github.com/stanford-oval/decaNLP.git#egg=decanlp'
```

Comment:
This part only installs the code dependencies: word embeddings are downloaded on first use. I wonder if it's more efficient to put the word embeddings in the docker image instead?
Downloading the embedding means downloading a zip file, then extracting a text file from it, and the preprocessing the file to make the .npy word embedding file that can mmap-ed; the latter is smaller and we should download it directly from https://parmesan.stanford.edu/glove/ like ./tests/install-nlp-deps.sh does.
(An alternative solution is to put the word embedding files in a shared NFS folder. Out research cluster does that, and it works well to save disk space, with very small overhead. S3 would not do though, because one would need a local copy anyway.)

Reply:
Does it take a long time to download from parmesan?  If everything is inside AWS network, it probably won't make too much difference unless Parmesan is overloader.  Putting embedding inside docker will make the image size larger and thus making it longer to download from ECR.  You can also use S3 if Paramesan is slow since S3 is provisioned for large traffic.

Reply:
Downloading from parmesan takes a while, but S3 costs money, while parmesan is a machine we own on the university network, which is free (subsidized). Inbound network traffic is also free in AWS.

Reply:
S3 storage should be really cheap. Looks like it's $0.023 per GB a month. 
https://aws.amazon.com/s3/pricing/
I believe transfer cost in the same region is free like in gcs. 

If you we want to speed up the download, it might be worthwhile to look into since transfer within the same network should be really fast and  S3 should also supports parallel download.